### DISCUSSION TITLE
WIll work in mobile apple devices?

### DISCUSSION THREAD

**Discussion by jalberto** [LBF38/obsidian-syncthing-integration#133] [Q&A]
There is not syncthing app for ios/ipados

So I wonder if it could remotely use synching API instead of relaying on a local installation.

**Comment by LBF38** [comment]
Hi @jalberto, as long as the Syncthing API is available at a specific address using internet, the plugin should be able to get the information and Syncthing's configuration.

By default, the address is `localhost:8384`. So, I suppose that it would work if you change it to your server where Syncthing is exposed.
It is just a call to API endpoints.

However, I can't provide any real feedback as I don't have any iPhone, so I rely on your feedbacks to know if it works. 

> **Note**
> For the moment, there are some HTTPS certificate issues on mobile. 
> So, it is possible that some plugin features aren't working yet.
> I'm still working on the overall plugin.

**Reply by tobiasb** [reply]
I quickly tested this on iOS. I had the same idea, instead of using the local Syncthing client (there's no such thing on iOS) I exposed my Syncthing "server" and configured this plugin to use that URL + API key.

When I click "Check API status" the banner shows "Syncthing Ping : pong", so far so good. When I click on "Syncthing Configuration" I see the Syncthing GUI, albeit a bit mangled but it works. What should I do from here? How do I select the Syncthing "folder" to use to sync my iOS local Obsidian vault with?

I understand that this is more of a general "How do I configure this plugin" kind of question now I think but I couldn't find anything in the README.

**Reply by LBF38** [reply]
The plugin aims to integrate most features of Syncthing in Obsidian. However, it's primarly designed to be used in a folder configured with Syncthing. And it works in pairs with a locally installed Syncthing instance. (The plugin doesn't contain or instantiate Syncthing).

If I understand correctly your usecase, you would like to use the plugin to retrieve a folder from your Syncthing installation through the plugin in another Obsidian Vault, am I correct ?

If correct, then I can say that, for the moment, all features available from the Syncthing GUI isn't yet fully integrated in the plugin. I'm making a big refactoring step for the remote datasource (calls to Syncthing REST API). Please see the state of the PR #137 for more info.

Hope it answers your question.

**Reply by tobiasb** [reply]
> If I understand correctly your usecase, you would like to use the plugin to retrieve a folder from your Syncthing installation through the plugin in another Obsidian Vault, am I correct ?

I think I'm misunderstanding what this plugin's goal is.

> The integration would interact with Syncthing using the CLI or the REST API. And the main objective is to handle all the vault synchronization process from the Obsidian app. Therefore, you don't have to open the Syncthing app on your devices to synchronize your vault.

Basically all this plugin should do is trigger the Syncthing sync whenever files change? On Android, is that even necessary? I used Syncthing+Obsidian for a while and I don't think I ever had to manually open the Syncthing app and sync.

Anyway if that's the case then this won't work for iOS. There is no "Syncthing" app, we only have Möbius which is Syncthing wrapped in an app that makes sure that the files are synced regularly because, I think, of how iOS works and how apps run in the background. Even if there was a way to talk to the local Syncthing service inside Möbius it still wouldn't work because iOS won't allow cross app file access the way Android does.

Let me know if I misunderstood anything. Thanks for your help.

**Reply by LBF38** [reply]
_Sorry for the delay in the answer. Been busy w/ other things._

I apologize if the goal of the plugin isn't clear. I have made the README before creating and building the whole plugin, so it might be a bit off on some points. Let me explain the purpose I found in making this plugin.

I have mainly created the plugin to have a "secure" way to synchronize my vault and files across my devices (phone, tablet, computer).
Therefore, ideally, I wanted to have an Obsidian plugin that allows me to schedule a synchronization regularly, depending on user's settings. The idea then evolved into making notifications to remind the user to turn on the Syncthing instance, when the system detects that the vault is at X% out of sync. Finally, I still think on how to achieve it in a simple manner.

To answer you, for now, the plugin has the following features : 
- Conflict manager : 
   - Allow the user to see a modal in Obsidian to manage conflicting files from Syncthing synchronisation.
   - See the difference between two conflicting files
   - Quickly resolve the conflict using buttons (accept right / left / cancel)
   - Edit files in the conflict modal.
- Settings : 
   - Basic settings to connect the plugin to the Syncthing instance running on the computer, tablet or phone.
   - (wip) A configuration modal to see the whole configuration of the connected Syncthing instance inside Obsidian.
   - A button to go to the Syncthing GUI (on the web).
- (wip) Status bar
   - A status bar to notify the user on the current state of the opened file. (see #11 and #138)
- And more to come.. (if I have time 😆)

So, some features specific to Obsidian such as the conflict modal can work without the Syncthing instance. However, if you want to have the configuration modal in Obsidian, you will need a running Syncthing instance on your iOS or Mac device. It only needs to expose the REST API.

**The plugin is only an integration of the Syncthing GUI with more features specific to handling files in Obsidian.** It gets information mainly through the REST API and, sometimes, through the CLI, only on desktop.

Hope it clarifies the goals of the plugin. 
And thank you for the discussion, as it helps me improve the plugin in a useful way.

**Reply by jalberto** [reply]
Thanks for your clear explanation and work, the plugin is awesome.

Sadly it will not work in ios/ipad devices as there is not syncthing client for their OS, hence the question about using a remote syncthing instance through the API even if the functionality is limited.

**Reply by KuiyueRO** [reply]
https://www.mobiussync.com/

**Reply by KuiyueRO** [reply]
A paid syncthing ios client that allows the use of obsidian sandbox folders

**Reply by tobiasb** [reply]
@KuiyueRO In Obsidian, how do you access the syncthing folder within Möbius Sync?

**Reply by KuiyueRO** [reply]
You can see pick external folder under the folder path, maybe you need to pay first.

**Reply by jalberto** [reply]
@KuiyueRO mobius is not available in mostly European countries

**Reply by jeslinmx** [reply]
I have multiple iOS devices with Mobius Sync, as well as other devices with Syncthing. A plugin which can control and trigger Syncthing from within Obsidian itself would be amazing as we would no longer be at the mercy of Mobius's intermittent syncing thanks to iOS's power management, which leads to all sorts of wonderful sync conflicts.

Please let me know in what way I can assist with testing and development, I also have some technical background and can write PRs if needed.

**Reply by LBF38** [reply]
@jeslinmx Thx for your message ! 
For the moment, the plugin relies on Syncthing's REST API and CLI (on desktop only). It doesn't sync anything at all and relies on a Syncthing instance such as the Android/iOS app. However, it could be interesting to test if we can integrate a Syncthing instance in the plugin (if not too heavy nor too complicated).

If you have time and you want to dive in, I think this would be helpful for iOS users, as said in this thread.
